checks/helm.go

```diff
@@ -0,0 +1,207 @@
+package checks
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/flanksource/canary-checker/pkg"
+	pusher "github.com/chartmuseum/helm-push/pkg/chartmuseum"
+	log "github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/cli"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/user"
+	"path"
+	"time"
+)
+
+type HelmChecker struct{}
+
+type ResultWriter struct {}
+
+// Type: returns checker type
+func (c *HelmChecker) Type() string {
+	return "helm"
+}
+
+func (c *HelmChecker) Run(config pkg.Config, results chan *pkg.CheckResult) {
+	for _, conf := range config.Helm {
+		results <- c.Check(conf.HelmCheck)
+	}
+}
+
+func (c *HelmChecker) Check(config pkg.HelmCheck) *pkg.CheckResult {
+	start := time.Now()
+	var uploadOK, downloadOK bool = true, true
+	chartmuseum := fmt.Sprintf("%s/chartrepo/%s/", config.Chartmuseum, config.Project)
+	log.Trace("Uploading test chart")
+	client, _  := pusher.NewClient(
+		pusher.URL(chartmuseum),
+		pusher.Username(config.Username),
+		pusher.Password(config.Password),
+		pusher.ContextPath(""),
+		pusher.Timeout(60),
+		pusher.CAFile(*config.CaFile),)
+	response, err := client.UploadChartPackage("/Users/rubenharutyunov/Upwork/flanksource/canary_checker/fixtures/test-chart-0.1.0.tgz", false)
+
+	if err != nil {
+		return &pkg.CheckResult{
+			Pass:     false,
+			Invalid:  true,
+			Duration: 0,
+			Endpoint: config.Chartmuseum,
+			Message:  fmt.Sprintf("Failed to check: %v", err),
+			Metrics:  getHelmMetrics(config, false),
+		}
+	}
+	log.Trace(response)
+	log.Trace(response.Request)
+
+	if response.StatusCode != 201 {
+		uploadOK = false
+		return &pkg.CheckResult{
+			Pass:     false,
+			Invalid:  false,
+			Duration: 0,
+			Endpoint: config.Chartmuseum,
+			Message:  "Failed to push test chart",
+			Metrics:  getHelmMetrics(config, false),
+		}
+	}
+
+	user, err := user.Current()
+	if err != nil {
+		return &pkg.CheckResult{
+			Pass:     false,
+			Invalid:  true,
+			Duration: 0,
+			Endpoint: config.Chartmuseum,
+			Message:  fmt.Sprintf("Failed to get user: %v", err),
+			Metrics:  getHelmMetrics(config, false),
+		}
+	}
+	kubeconfigPath := path.Join(user.HomeDir, ".kube/config")
```

Comment:
move this closer to its first use - also use the logic from here: https://github.com/flanksource/canary-checker/blob/master/pkg/kube.go#L15